nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -133,6 +133,7 @@ let
       in
         if x == null then "false"
         else boolToString x;
+    dropNullFields = filterAttrs (_: v: v != null);
```

Comment:
This seems like a way to introduce very weird and unexpected behavior. I think it's preferable to explicitly set defaults here anyways (then these are also documented in the manual).

I think I'd prefer to explicitly drop dbuser/dbname if dbtype=="sqlite" and vice versa (also, an assertion should be thrown if e.g. sqlite & dbuser are set rather than silently letting this fail at runtime).

Reply:
> I think I'd prefer to explicitly drop dbuser/dbname if dbtype=="sqlite" and vice versa (also, an assertion should be thrown if e.g. sqlite & dbuser are set rather than silently letting this fail at runtime).

Thanks for the feedback :) How should this work exactly? Does one have to explicitly set dbuser/dbname to `null` if dbtype=="sqlite"?